src/core/gui/XournalView.cpp

```diff
@@ -681,28 +685,33 @@ void XournalView::repaintSelection(bool evenWithoutSelection) {
     gtk_widget_queue_draw(this->widget);
 }
 
-void XournalView::setSetsquareView(std::unique_ptr<SetsquareView> setsquareView) {
-    GTK_XOURNAL(this->widget)->setsquareView = std::move(setsquareView);
-}
+auto XournalView::getGeometryToolHandler() -> GeometryToolHandler* { return geometryToolHandler; }
 
-void XournalView::resetSetsquareView() {
-    GTK_XOURNAL(this->widget)->setsquareView.reset();
-    this->control->fireActionSelected(GROUP_SETSQUARE, ACTION_NONE);
-}
+auto XournalView::getGeometryToolController() -> GeometryToolController* { return geometryToolController; }
 
-auto XournalView::getSetsquareView() const -> SetsquareView* {
-    g_return_val_if_fail(this->widget != nullptr, nullptr);
-    g_return_val_if_fail(GTK_IS_XOURNAL(this->widget), nullptr);
+void XournalView::makeGeometryTool(std::string tool) {
+    auto view = getViewFor(control->getCurrentPageNo());
+    auto* zoomControl = this->getControl()->getZoomControl();
 
-    return GTK_XOURNAL(this->widget)->setsquareView.get();
+    if (tool == "setsquare") {
+        auto setsquare = new Setsquare();
+        view->addOverlayView(std::make_unique<xoj::view::SetsquareView>(setsquare, view, zoomControl));
+        this->geometryTool = setsquare;
+        this->geometryToolController = new SetsquareController(view, setsquare);
+        this->geometryToolHandler = new SetsquareInputHandler(this, this->geometryToolController);
```

Comment:
The controller should be owned by the Control instance. Not sure about the InputHandler. Also, the name `geometryToolHandler` is confusing. The distinction between controller and handler is not clear to me...

Overall, my strategy for determining "who owns what" is: ask yourself what will happen if and when we finally split PageView and XournalView into an MVC pattern. In this instance, it's not quite clear yet but it's a good thing to keep in mind.

Also, whoever owns them, they should be handled with a smart pointer.

Reply:
`geometryToolHandler` could be renamed to `geometryToolInputHandler` if that's less confusing. The input handler is just for processing input. It asks the controller to actually control (move, scale, rotate ...) the model (geometry tool).
When migrating to Gtk4 we will just to modify the input handler, not the controller.

Reply:
Yes, this would be less confusing I guess.

Reply:
Renaming is done.

Reply:
> The controller should be owned by the Control instance. Not sure about the InputHandler. Also, the name `geometryToolHandler` is confusing. The distinction between controller and handler is not clear to me...
> 
> Overall, my strategy for determining "who owns what" is: ask yourself what will happen if and when we finally split PageView and XournalView into an MVC pattern. In this instance, it's not quite clear yet but it's a good thing to keep in mind.
> 
> Also, whoever owns them, they should be handled with a smart pointer.

The `geometryToolController` is now owned by the `Control` instance, the `geometryToolInputHandler` by the `InputContext` instance. That makes the most sense to me, as they are now where they are used.

Reply:
The `geometryTool` is now owned by the `Control` instance. I see no reason to pass it forth and back.